### PR TITLE
Fix for linter issues seen while building with the latest golang and linter

### PR DESCRIPTION
**Pull request by rohit-balakrishna**
Built with go version go1.23.1 and golangci-lint version 1.61.0 built with go1.23.1

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -163,3 +163,5 @@ replace k8s.io/sample-controller => k8s.io/sample-controller v0.25.16
 replace k8s.io/kube-openapi => k8s.io/kube-openapi v0.0.0-20230525220651-2546d827e515
 
 replace google.golang.org/protobuf => google.golang.org/protobuf v1.33.0
+
+replace github.com/hpe-storage/common-host-libs => ../common-host-libs
```
Comment (rohit-balakrishna) [inline go.mod:167]: Will remove this once the common-host-libs change is accepted.

**Comment by datamattsson** [comment]
This PR passes e2e tests.

```
[ReportAfterSuite] PASSED [0.135 seconds]
------------------------------

Ran 90 of 6897 Specs in 3963.523 seconds
SUCCESS! -- 90 Passed | 0 Failed | 0 Pending | 6807 Skipped
PASS
```

File: vendor/modules.txt [hunk 2]
```diff
@@ -113,7 +113,7 @@ github.com/google/uuid
 # github.com/gorilla/mux v1.8.0
 ## explicit; go 1.12
 github.com/gorilla/mux
-# github.com/hpe-storage/common-host-libs v0.0.0-20240717051902-d204e2e2e8fd
+# github.com/hpe-storage/common-host-libs v0.0.0-20240717051902-d204e2e2e8fd => ../common-host-libs
 ## explicit; go 1.19
```
Comment (dileepds) [inline vendor/modules.txt:117]: The vendor related files should always refer to the github.com path instead of the local path.

File: go.mod (hunk 1 above)
Comment (rohit-balakrishna) [reply go.mod]: Fixed

**Review by dileepds — APPROVED** [review]
LGTM

**Review by sijeesh — APPROVED** [review]
LGTM

**Pull request merged**